src/engine/strat_engine/thinpool/filesystem.rs

```diff
@@ -157,27 +159,18 @@ impl StratFilesystem {
     /// system that is contained on the block device referred to as self.devnode(), i.e. the device
     /// node, while ignoring parse errors as long as at least one mount point is found.
     pub fn get_mount_point(&self) -> StratisResult<Option<PathBuf>> {
-        let device_node = self.devnode();
-        let search = device_node.to_str().ok_or_else(|| {
-            StratisError::Engine(
-                ErrorEnum::Error,
-                format!("Unable to represent devnode as string {:?}", *self),
-            )
-        })?;
+        let search = self.devnode().into_os_string();
 
-        let m_iter = MountIter::new_from_proc().map_err(|e| {
-            StratisError::Engine(
-                ErrorEnum::Error,
-                format!("Error reading /proc/mounts {:?}", e),
-            )
-        })?;
+        let mut mount_data = String::new();
+        File::open("/proc/self/mountinfo")?.read_to_string(&mut mount_data)?;
+        let parser = libmount::mountinfo::Parser::new(mount_data.as_bytes());
 
         let mut last_error: Option<String> = None;
-        for mp in m_iter {
+        for mp in parser.into_iter() {
             match mp {
                 Ok(mount) => {
-                    if mount.contains(&MountParam::Spec(search)) {
-                        return Ok(Some(mount.file));
+                    if mount.mount_source == *search {
```

Comment:
Yeah, I think we just want to use maj/min for comparison here, is all. Something like:
```
let major = self.thin_dev.device().major as u64;
let minor = self.thin_dev.device().minor as u64;
if mount.major == major && mount.minor == minor {
```